#! /bin/sh
patch -p1 -f $* < $0
exit $?

2006-01-01  Tor Lillqvist  <tml@novell.com>

	* glib/gspawn-win32.c (g_spawn_sync_utf8): Set the GIOChannels for
	stdout and stderr to unbuffered. Otherwise the giochannel layer
	will try to read from them regardless whether the
	g_io_channel_win32_poll() call here has indicated
	readability or not. (#325310)

--- glib-2.9.2-wk1/glib/gspawn-win32.c	2006-05-28 21:34:46.000000000 +0200
+++ glib/glib/gspawn-win32.c	2006-05-28 21:34:55.000000000 +0200
@@ -1058,9 +1035,12 @@
       outstr = g_string_new (NULL);
       outchannel = g_io_channel_win32_new_fd (outpipe);
       g_io_channel_set_encoding (outchannel, NULL, NULL);
+      g_io_channel_set_buffered (outchannel, FALSE);
       g_io_channel_win32_make_pollfd (outchannel,
 				      G_IO_IN | G_IO_ERR | G_IO_HUP,
 				      &outfd);
+      if (debug)
+	g_print ("outfd=%x\n", outfd.fd);
     }
       
   if (errpipe >= 0)
@@ -1068,9 +1048,12 @@
       errstr = g_string_new (NULL);
       errchannel = g_io_channel_win32_new_fd (errpipe);
       g_io_channel_set_encoding (errchannel, NULL, NULL);
+      g_io_channel_set_buffered (errchannel, FALSE);
       g_io_channel_win32_make_pollfd (errchannel,
 				      G_IO_IN | G_IO_ERR | G_IO_HUP,
 				      &errfd);
+      if (debug)
+	g_print ("errfd=%x\n", errfd.fd);
     }
 
   /* Read data until we get EOF on all pipes. */
